### DISCUSSION TITLE
Missing days in SpotData table in SQLite DB

### DISCUSSION THREAD

**Discussion by shutterfreak** [SBFspot/SBFspot#629] [Q&A] [answered]
I have about 10 days of data missing in the SQLite DB. At least, the data points don't appear in the SpotData table.

Is there a way to retrieve and add these missing data points?

Is there a possibility to retrieve all historical data points from an inverter (besides retrieving them as text files through Sunny Explorer)?

**Comment by sbf-** [comment]
> Is there a way to retrieve and add these missing data points

No, spotdata is not kept in the inverter. The data is lost.

> Is there a possibility to retrieve all historical data points from an inverter

Yes, use the `-ad` and `-am` options.
For example, to retrieve daydata af last week and monthdata of 2 months, use `- ad7 -am2`

`-?` shows all possible arguments

**Comment by shutterfreak** [comment, marked as answer]
I see that SBFspot is able to retrieve the production data for the missing days from the inverter but it won't add these records to the SpotData table in the SQLite database. Probably because it also wants to keep track of other parameters which aren't kept in the inverter?

The data records retrieved by running SBFspot and reported in the daydata overview of that day contain the timestamp of retrieval, cumulated total power (ETotal, kWh) and instant power (Pac1, W). So I can reconstruct the missing data and add it to a database, although it requires some editing to create the proper queries (and tweaking the clock if the timezone differs).

**Reply by sbf-** [reply]
> I see that SBFspot is able to retrieve the production data for the missing days from the inverter but it won't add these records to the SpotData table in the SQLite database

The 5min historical data is stored in `daydata` table and totals per month in `monthdata` table
`spotdata` table is for instant data like actual ac/dc voltages, ac/dc power...
It is storing one record at each run of SBFspot

> So I can reconstruct the missing data and add it to a database, although it requires some editing to create the proper queries (and tweaking the clock if the timezone differs).

I really can't imagine what you're trying to do, sorry.

**Reply by shutterfreak** [reply]
Thanks! I now understand which data are stored in which table.

I was trying to expose the SPFspot data to openHAB and wanted to bridge data gaps.

Thanks to your explanation it was a straightforward thing to do!
